File: Luogu/Greedy/P3817.cc
/* P3817.cc */

#include <iostream>
#include <vector>

int main(void) {
    long long n = 0, x = 0, candy = 0, sum = 0;
    std::cin >> n >> x;

    std::vector<long long> candies(n, 0);

    std::cin >> candies[0];
    if (x < candies[0]) {
        sum += (candies[0] - x);
        candies[0] = x;
    }

    for (long long i = 1; i < n; i ++) {
        std::cin >> candies[i];
        if (x < candies[i] + candies[i - 1]) {
            candy = x - candies[i - 1];
            sum += (candies[i] - candy);
            candies[i] = candy;
        }
    }

    std::cout << sum << std::endl;

    return 0;
}

File: Luogu/Greedy/P3817.py
nums = []
candies = []
in_str = list(input().split(" "))
sum = 0
for str in in_str:
    if ('' != str):
        nums.append(int(str))
n = nums[0]
x = nums[1]

in_str = list(input().split(" "))
for str in in_str:
    if ('' != str):
        candies.append(int(str))

if (candies[0] > x):
    sum += candies[0] - x
    candies[0] = x

for i in range(1, n):
    if (candies[i - 1] + candies[i] > x):
        sum += (candies[i - 1] + candies[i] - x)
        candies[i] = (x - candies[i - 1])

print(sum)